github_graphql.go

```diff
@@ -47,7 +49,17 @@ func (g *GitHubClient) listReleasesV4() ([]*github.RepositoryRelease, error) {
 			r := r
 			publishedAt, _ := time.ParseInLocation(time.RFC3339, r.Node.PublishedAt.Time.Format(time.RFC3339), time.UTC)
 			createdAt, _ := time.ParseInLocation(time.RFC3339, r.Node.CreatedAt.Time.Format(time.RFC3339), time.UTC)
-			releaseID, _ := strconv.ParseInt(r.Node.ID, 10, 64)
+			var releaseID int64
+			decodedID, err := base64.StdEncoding.DecodeString(r.Node.ID)
+			if err != nil {
+				return nil, err
+			}
+			re := regexp.MustCompile(`.*[^\d]`)
+			decodedID = re.ReplaceAll(decodedID, []byte(""))
+			releaseID, err = strconv.ParseInt(r.Node.ID, 10, 64)
```

Comment:
I don't think this'll quite work - the base64 looks like `07:Release33356981`, so if we just remove all non-digits, we'll end up trying to parse `0733356981` rather than `33356981`. Maybe something like this instead?

```suggestion
			re := regexp.MustCompile(`(\d+)$`)
			match := re.FindStringSubmatch(decodedID)
			if len(match) < 2 {
				return nil, fmt.Errorf("unexpected release ID %q", decodedID)
			}
			releaseID, err = strconv.ParseInt(match[1], 10, 64)
```

Reply:
Checked it's working now, I forgot to correctly use variables @aoldershaw please check now, also I adjusted one test to match error with the current changed behavior